### PR TITLE
boards/nucleo-l011k4: add support

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR adds support to the nucleo-l011k4 which is the smallest nucleo boards available: only 2kB of RAM and 16kB of flash memory.

To be able to use it, PICOLIBC must be enabled and the default thread stack size must be reduced to 512B.

Tested with success: UART, ADC, RTC, RTT

### Testing procedure

- A green Murdock (it might fail because of picolibc requirement)
- Run `BUILD_IN_DOCKER=1 ./dist/tools/compile_and_test_for_board/compile_and_test_for_board.py . nucleo-l011k4 --jobs=8 --with-test-only`

```
- [tests/driver_apds99xx](tests/driver_apds99xx/test.failed)
- [tests/driver_apds99xx_full](tests/driver_apds99xx_full/test.failed)
- [tests/driver_grove_ledbar](tests/driver_grove_ledbar/test.failed)
- [tests/driver_my9221](tests/driver_my9221/test.failed)
- [tests/periph_timer_short_relative_set](tests/periph_timer_short_relative_set/test.failed)
```

### Issues/PRs references

None

### CODE REVIEW

**Comment by aabadie** [comment]
Now that #15011 is merged, the build is broken with nucleo-l011k4. I'm looking into it.

**Comment by aabadie** [comment]
> Now that #15011 is merged, the build is broken with nucleo-l011k4

For the moment, I have no idea how to fix this problem. The only place where picolibc can be pulled in is from the application Makefile (so before anything is done by the build system).
Setting a strong dependency to picolibc at board level is not possible because some newlib modules are already pulled-in at that moment. That's a mess...

File: cpu/cortexm_common/Makefile.dep [hunk 1]
```diff
@@ -4,7 +4,7 @@ USEMODULE += cortexm_common
 # include common periph code
 USEMODULE += cortexm_common_periph
 
-ifneq (,$(filter picolibc,$(FEATURES_USED)))
+ifneq (,$(filter picolibc,$(FEATURES_USED) $(USEMODULE)))
```
Comment (benpicco) [inline cpu/cortexm_common/Makefile.dep:7]: Doesn't this mean we will *always* use `picolibc` now?

I mean I don't object, but this should probably be discussed in another PR :wink:

Comment (benpicco) [reply cpu/cortexm_common/Makefile.dep:7]: Wait, I have it reversed: if `picolibc` is in `$(FEATURES_USED)` or `$(USEMODULE)`, `picolibc` is added to `USEMODULE` - how does this make a difference? :exploding_head:

Comment (aabadie) [reply cpu/cortexm_common/Makefile.dep:7]: It means that if picolibc is already in USEMODULE then don't use newlib. When using FEATURES_REQUIRED in the board Makefile.dep, picolibc is not yet added to FEATURES_USED at this stage so newlib is added the first time this Makefile is parsed.
It doesn't mean picolibc is always used.

Comment (aabadie) [reply cpu/cortexm_common/Makefile.dep:7]: The other solution is to skip this whole block if picolibc is already in USEMODULE but I'm not sure that makes sense.

Comment (benpicco) [reply cpu/cortexm_common/Makefile.dep:7]: Ah now it's dawning upon me: `FEATURES_REQUIRED` gets evaluated to late, that's why first the `newlib` path is taken.

That`s why you changed it to

    USEMODULE += picolibc

Comment (aabadie) [reply cpu/cortexm_common/Makefile.dep:7]: It took me a bit of time to figure that out :)

Comment (benpicco) [reply cpu/cortexm_common/Makefile.dep:7]: Can you also add it to `cpu/fe310` for consistency.
I hope Kconfig will soon redeem us from this (*right?* *right?*).

Comment (aabadie) [reply cpu/cortexm_common/Makefile.dep:7]: > Can you also add it to cpu/fe310 for consistency.

I'm not sure, it's very unlikely that a board will have to force the use of picolibc in order to work. Once dependency resolution is managed by Kconfig, we won't need this anymore I hope.

Comment (benpicco) [reply cpu/cortexm_common/Makefile.dep:7]: Then at least add a comment as to why this is needed - you can squash directly.

File: examples/lua_REPL/Makefile.ci [hunk 2]
```diff
@@ -1,3 +1,7 @@
+BOARD_BLACKLIST := \
+    nucleo-l011k4 \
```
Comment (benpicco) [inline examples/lua_REPL/Makefile.ci:2]: Why not `BOARD_INSUFFICIENT_MEMORY`?

File: examples/lua_basic/Makefile.ci [hunk 3]
```diff
@@ -1,3 +1,7 @@
+BOARD_BLACKLIST := \
+    nucleo-l011k4 \
```
Comment (benpicco) [inline examples/lua_basic/Makefile.ci:2]: This *will* get deleted by the next run of `add_insufficient_memory_board.sh`.

File: examples/lua_REPL/Makefile.ci (hunk 2 above)
Comment (aabadie) [reply examples/lua_REPL/Makefile.ci]: Ok, I moved this to the main Makefile with a comment (same for `lua_basic`).

**Review by benpicco — APPROVED** [review]
